src/app/shared/error/global-error-handler.ts

```diff
@@ -2,23 +2,42 @@ import { NotificationService } from './../notification/notification.service';
 import { HttpErrorResponse } from '@angular/common/http';
 import { ErrorHandler, Injectable } from '@angular/core';
 import { ErrorService } from './error.service';
+import { ProblemResponse } from 'api-client';
 
 @Injectable({
   providedIn: 'root',
 })
 export class GlobalErrorHandler implements ErrorHandler {
+  constructor(private errorService: ErrorService, private notifier: NotificationService) { }
 
-    constructor(private errorService: ErrorService, private notifier: NotificationService) { }
-
-    handleError(errorResponse: Error | HttpErrorResponse) {
-        let message: string;
+  handleError(errorResponse: Error | HttpErrorResponse) {
+    let message: string;
+    if (errorResponse instanceof HttpErrorResponse) {
+      if (this.isProblemResponse(errorResponse)) {
+        this.notifier.showProblem(this.makeProblem(errorResponse.error));
+        return;
+      } else {
+        message = errorResponse.message;
+      }
+    } else {
+      message = this.errorService.getClientMessage(errorResponse);
+    }
+    this.notifier.showError(message);
+  }
 
-        if (errorResponse instanceof HttpErrorResponse) {
-          message = errorResponse.message;
-        } else {
-          message = this.errorService.getClientMessage(errorResponse);
-        }
+  private isProblemResponse(errorResponse: HttpErrorResponse): boolean {
+    return errorResponse.headers.get('Content-Type') === 'application/problem+json';
+  }
 
-        this.notifier.showError(message);
-    }
+  private makeProblem(error: any): ProblemResponse {
+    const { detail, errors, instance, status, title, type } = error;
+    return {
+      detail,
+      errors,
+      instance,
+      status,
+      title,
+      type,
+    };
+  }
```

Comment:
nit: Unconventional term `make` in TS code. Suggest perhaps `createProblem` or `createProblemFromError` to be more descriptive